src/bin/pred.rs

```diff
@@ -276,12 +276,20 @@ impl SynthLanguage for Pred {
             env.insert(var, vec![]);
         }
 
-        for cvec in env.values_mut() {
+        for (var, cvec) in env.iter_mut() {
             cvec.reserve(n * 3);
             let mut vals = vec![];
-            vals.append(&mut sampler(&mut synth.rng, 8, 6, n));
-            vals.append(&mut sampler(&mut synth.rng, 6, 8, n));
-            vals.append(&mut sampler(&mut synth.rng, 3, 1, n));
+            if var.to_string().starts_with("?b") {
+                for _ in 0..(n * 3) {
+                    vals.push(Constant::Bool(synth.rng.gen::<bool>()));
```

Comment:
the bool domain just does a cross product of `True and False` over the three variables. Wondering if we should reuse that idea or generate at random like here. Did you look at that and decide against it? Either way is fine, I am just wondering.